server/controllers/api_controller.go

```diff
@@ -92,7 +92,7 @@ func (a *APIController) Plan(w http.ResponseWriter, r *http.Request) {
 		a.apiReportError(w, http.StatusInternalServerError, err)
 		return
 	}
-	defer a.Locker.UnlockByPull(ctx.HeadRepo.FullName, 0) // nolint: errcheck
+	defer a.Locker.UnlockByPull(ctx.HeadRepo.FullName, 0, true) // nolint: errcheck
 	if result.HasErrors() {
```

Comment:
Have you considered gating this functionality behind a feature flag in case users want to opt-out?

Reply:
Agreed. Ok, let's put this behind a flag as a prereq prior to merging this PR.

We would then need to test end to end without plan queueing and with plan queueing to ensure nothing breaks and people can opt into this feature.

From your comments, it sounds like you may not have the time for this, so we welcome any additional contributors to help get that over the line.

Reply:
@nitrocode I added the feature flag in 6937973, next I will add end to end tests for the feature flag and resolve other comments.